e2e/app/page-mixin/SideNav.ts

```diff
@@ -0,0 +1,83 @@
+import {Page} from 'puppeteer';
+import {iconXpath} from '../aou-elements/xpath-defaults';
+import {findIcon} from '../aou-elements/xpath-finder';
+
+export interface PuppeteerPage {
+  page: Page;
+}
+
+type Constructor<T> = new(...args: any[]) => T;
+
+// @ts-ignore
```

Comment:
I don't think this mixin pattern is going to work well in Typescript as is and I'm not sure it's buying much. I would suggest using a composition based approach instead.

```
class Data extends AuthenticatedPage {
  public sidenav: SideNav;
}

...

test() {
  dataPage.sidenav.goTo(...)
}
```

If all authenticated pages have a side nav, move it to the base class instead

Reply:
mixin is gone. functions moved to `BasePage`

Reply:
felt satisfying write very confusing and challenging JS code. 😉